.github/workflows/ci.yml

```diff
@@ -39,10 +39,12 @@ jobs:
         - { name: Linux Clang,                    os: ubuntu-22.04, flags: -DCMAKE_CXX_COMPILER=clang++ -DSFML_RUN_DISPLAY_TESTS=ON -GNinja , gcovr_options: '--gcov-executable="llvm-cov-$CLANG_VERSION gcov"' }
         - { name: Linux GCC DRM,                  os: ubuntu-22.04, flags: -DSFML_USE_DRM=ON -DSFML_RUN_DISPLAY_TESTS=OFF -GNinja }
         - { name: Linux GCC OpenGL ES,            os: ubuntu-22.04, flags: -DSFML_OPENGL_ES=ON -DSFML_RUN_DISPLAY_TESTS=ON -GNinja }
-        - { name: macOS,                          os: macos-12, flags: -GNinja }
-        - { name: macOS Xcode,                    os: macos-12, flags: -GXcode }
-        - { name: iOS,                            os: macos-12, flags: -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_OSX_ARCHITECTURES=arm64 }
-        - { name: iOS Xcode,                      os: macos-12, flags: -DCMAKE_SYSTEM_NAME=iOS -GXcode -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO }
+        - { name: macOS x64,                      os: macos-14, flags: -GNinja -DCMAKE_OSX_ARCHITECTURES=x86_64 }
```

Comment:
If all macOS 14 images are using ARM then I'd rather we just use the macOS 14 jobs to test ARM and macOS 13 jobs to test x86. That further deduplicates things and lets us skip setting CMAKE_OSX_ARCHITECTURES since we'll always use the native architecture.

Reply:
Eventually `macos-13` will go away, but we can also stick to it for as long as it exists. I kind of like that flag, as it makes it explicit what's going on, rather than relying on hidden knowledge that `macos-13` is x64 based and `macos-14` is arm64 based.

Reply:
We can use the CI job name to make it clear what's going on. I find that more expressive than adding unnecessary parameters.